app/static/js/controllers/notificationController.js

```diff
@@ -93,5 +94,17 @@ function ($scope, $rootScope, $filter, Cluster) {
       item.customTimeStamp =  new Date(item.eventTimestamp).toLocaleDateString() + " " + new Date(item.eventTimestamp).toLocaleTimeString();
       $rootScope.events.push(item);
     }
+    
+    function refreshMetadata(notification) {
+       GlobalStack.get({ id: notification.stackId }, function(success) {
+          var metadata = []
+          angular.forEach(success.instanceGroups, function(item) {
+             angular.forEach(item.metadata, function(item1) {
+               metadata.push(item1)
+             });
+             $rootScope.activeCluster.metadata = metadata // trigger activeCluster.metadata
```

Comment:
This line should be out of the foreach scope, because we would not like to refresh metadata for instance groups individually.